### DISCUSSION TITLE
FEATURE REQUEST: Mad Max Support

### DISCUSSION THREAD

**Discussion by Xefier** [swar/Swar-Chia-Plot-Manager#1216] [Ideas]
Would be great if this supported Mad Max Plotter which is much faster
https://github.com/madMAx43v3r/chia-plotter

**Comment by kianschki** [comment]
Mad Max is his own plotting manager. No need to integrate it. The design is a diffrent one. It is a pipepline which puts all the power on **one** plot. This way it can create a plot in under one hour. Sometimes in about ten minutes. Depending on the hardware. You will need a lot of RAM for a RAM drive and only 25% is storred on the nvme. So you create one plot after the other.

**Reply by CryptoBlockchainTechnologies** [reply]
Huge benefit to run MM with multiple parallel processes for those of us who have the beefy systems. Just need a system like Swars to stagger the plot start times, just like Swars does with the Chia CLI. This is what a lot of us are already doing but with custom scripting.

**Reply by Chilli71** [reply]
> Mad Max is his own plotting manager. No need to integrate it. The design is a diffrent one. It is a pipepline which puts all the power on **one** plot. This way it can create a plot in under one hour. Sometimes in about ten minutes. Depending on the hardware. You will need a lot of RAM for a RAM drive and only 25% is storred on the nvme. So you create one plot after the other.

I believe madmax is faster though.
It works almost like the origina chia plot function. Why should it be faster, when not replacing the chia plot function with an optimized code? You can use the chia plotting function with a RAM drive as well. There must be added value in MadMax.
I have a machine where I do 16plots in parallel with 16 temp drives and 16 temp2 drives.
MadMax is not usable in this configuration.
If it is faster than the genuine chia plotting function it would be worth a shot to integrate support into swar as well.

**Reply by delucca** [reply]
> Mad Max is his own plotting manager. No need to integrate it. The design is a diffrent one. It is a pipepline which puts all the power on **one** plot. This way it can create a plot in under one hour. Sometimes in about ten minutes. Depending on the hardware. You will need a lot of RAM for a RAM drive and only 25% is storred on the nvme. So you create one plot after the other.

That is not 100% true

A plot manager has plenty of use cases. You don't need to use a plot manager just for "parallelizing" plot creation. You could use it to simplify how you launch plot processes. That is even more required when you have a complex infrastructure.

Take my infrastructure, for example, I have 3 rigs. It is pretty annoying to configure bash scripts on all those rigs and change the script when I need to change the final HDD.

Although your idea is correct (Madmax is indeed designed for fast serial plotting), using a plot manager makes that process easier specially if you have more than one rig.

**Comment by swar** [comment]
MadMax support has been added thanks to @delucca's pull request

**Reply by 69chargr** [reply]
This is awesome but how to we use it besides making the switch in config.yaml
Thanks for your work!!

**Reply by Chilli71** [reply]
I would say at the parameter chia_location: specify the madmax location, at backend: write madmax and the rest stays the same.

**Reply by jonathanhatch** [reply]
Cool. Let's say I have 2TB of NVMe and 386 GB of Ram available, oh, and 64 cores. What would a decent config look like?

**Reply by Chilli71** [reply]
This is the wrong forum for that question! But with that amount of RAM I would say forget the NVMe and do everything in RAM. It might even be worth investigating if two staggered parallel plots in RAM are faster.
You can use the NVMe as buffer for your destination drive. Otherwise writng the plots to final destination would definitely be the bottle neck.
